src/MoBi.Core/Commands/RenameContainerCommand.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using MoBi.Assets;
+using MoBi.Core.Domain.Model;
+using OSPSuite.Assets;
+using OSPSuite.Core.Commands.Core;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Events;
+using OSPSuite.Core.Extensions;
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Core.Commands
+{
+   public class RenameContainerCommand : BuildingBlockChangeCommandBase<ISpatialStructure>
+   {
+      private IContainer _container;
+      private readonly string _newName;
+      private string _oldName;
+      private readonly string _containerId;
+
+      public RenameContainerCommand(IContainer container, string newName, ISpatialStructure spatialStructure) : base(spatialStructure)
+      {
+         _container = container;
+         _newName = newName;
+         _containerId = container.Id;
+         ObjectType = ObjectTypes.Container;
+         CommandType = AppConstants.Commands.RenameCommand;
+         Description = AppConstants.Commands.RenameDescription(_container, newName);
+      }
+
+      protected override ICommand<IMoBiContext> GetInverseCommand(IMoBiContext context)
+      {
+         return new RenameContainerCommand(_container, _oldName, _buildingBlock).AsInverseFor(this);
+      }
+
+      protected override void ClearReferences()
+      {
+         base.ClearReferences();
+         _container = null;
+      }
+
+      protected override void ExecuteWith(IMoBiContext context)
+      {
+         base.ExecuteWith(context);
+         var objectPathFactory = context.ObjectPathFactory;
+         var containerPath = objectPathFactory.CreateAbsoluteObjectPath(_container);
+
+         //perform the rename
+         _oldName = _container.Name;
+         _container.Name = _newName;
+
+         //the container has been renamed, we need to update the path of all neighborhoods connected to it
+         //no need to update referenced in the neighborhood as they have not changed
+         var updateNeighborhoods = updateNeighborhoodsDef(containerPath, objectPathFactory.CreateAbsoluteObjectPath(_container));
```

Comment:
@Yuri05 @rwmcintosh @georgeDaskalakis This is a new try for me in .NET to create curried function that capture a context so that you don't have to repeat yourself with parameter who do not change.
See now the next call
instead of having to do 
```
updateNeighborhoods(containerPath, newContainerPath, x => x.FirstNeighborPath)
updateNeighborhoods(containerPath, newContainerPath, x => x.SecondNeighborPath)
```

I can simply do 
```
updateNeighborhoods(x => x.FirstNeighborPath)
updateNeighborhoods(x => x.SecondNeighborPath)
```

and do not even need to create a local variable newContainerPath

Thoughts on this syntax? It's very common to do this in functional languages such as javascript and I think it can simply the code a bit (here maybe not but I just want to get the hang of it)